docs/DeveloperGuide.md

```diff
@@ -1,11 +1,233 @@
+---
+layout: default
+title: Developer Guide
+---
+* Table of Contents
+  {:toc}
+
+--------------------------------------------------------------------------------------------------------------------
+
 # Developer Guide
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+References
+1. Developer Guide: https://se-education.org/addressbook-level3/DeveloperGuide.html 
+2. User Guide: https://se-education.org/addressbook-level3/UserGuide.html
+3. 
+
 ## Design & implementation
 
+## Design
+
+### Architecture
+
+<img src="images/ArchitectureDiagram.png" width="280" />
+
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+
+Given below is a quick overview of main components and how they interact with each other.
```

Comment:
Perhaps a sequence diagram could show how all the components interact with each other?